src/main/java/com/microsoftopentechnologies/azure/AzureCloud.java

```
@@ -439,22 +486,12 @@ public int getDefaultMaxVMLimit() {
 			return Constants.DEFAULT_MAX_VM_LIMIT;
 		}
 
-		public FormValidation doVerifyConfiguration(@QueryParameter String subscriptionId, @QueryParameter String serviceManagementCert, 
-				@QueryParameter String passPhrase, @QueryParameter String serviceManagementURL) {
-			
-			if (AzureUtil.isNull(subscriptionId)) {
-				return FormValidation.error("Error: Subscription ID is missing");
-			}
-			
-			if (AzureUtil.isNull(serviceManagementCert)) {
-				return FormValidation.error("Error: Management service certificate is missing");
-			}
-			
-			if (AzureUtil.isNull(serviceManagementURL)) {
-				serviceManagementURL = Constants.DEFAULT_MANAGEMENT_URL;
-			}
-			
-			String response = AzureManagementServiceDelegate.verifyConfiguration(subscriptionId, serviceManagementCert, passPhrase, serviceManagementURL);
+		public FormValidation doVerifyConfiguration(@QueryParameter String credentialsId, @QueryParameter String passPhrase) {
+
+			AzurePublisherSettings credentials = resolveCredentials(credentialsId);
+
+			String response = AzureManagementServiceDelegate.verifyConfiguration(
```

Missing null check. Should return a form error IMO